ivy.el

```diff
@@ -4440,6 +4440,17 @@ BUFFER may be a string or nil."
           (find-file-other-window (cdr virtual))
         (switch-to-buffer-other-window buffer)))))
 
+(defun ivy--switch-buffer-other-tab-action (buffer)
+  "Switch to BUFFER in other tab.
+BUFFER may be a string or nil."
+  (if (zerop (length buffer))
+      (switch-to-buffer-other-tab ivy-text)
+    (let ((virtual (assoc buffer ivy--virtual-buffers)))
+      (if (and virtual
+               (not (get-buffer buffer)))
+          (find-file-other-tab (cdr virtual))
+        (switch-to-buffer-other-tab buffer)))))
```

Comment:
I pushed a commit to try and avoid this duplication between actions.  Let me know what you think.

Reply:
Looks good to me, it makes sense to abstract it a bit.